Avoid clang compiler warning

clang compiler is dumping a warning for socket.c code:

```
error: CLANG_WARNING:
tang-14/src/socket.c:184:10: warning[deadcode.DeadStores]: Although the value stored to 'pid' is used in the enclosing expression, the value is never actually read from 'pid'
#  182|   	int status;
#  183|   
#  184|-> 	while ((pid = waitpid(-1, &status, WNOHANG)) > 0);
#  185|   }
#  186|
```

This can be easily fixed, as pid variable is not being used

--- a/src/socket.c
+++ b/src/socket.c
@@ -178,10 +178,9 @@ static void spawn_process(int fd, const char *jwkdir,
 
 static void handle_child(int sig)
 {
-	pid_t pid;
 	int status;
 
-	while ((pid = waitpid(-1, &status, WNOHANG)) > 0);
+	while ((waitpid(-1, &status, WNOHANG)) > 0);
 }
 
 int run_service(const char *jwkdir, int port, process_request_func pfunc)

